### PR TITLE
Support optional access logs

### PR DESCRIPTION
**Pull request by brainsik**
This came up while I was drafting what a Hello, World project might look like. I don't want the student to have to create an S3 bucket + policy just to bring up an ALB. This felt like a simple change that doesn't impact the default of logging.

### CODE REVIEW

**Review by eeeady — CHANGES_REQUESTED** [review]
I just realized ... do you really want to disable logs??? I think that's a weird pattern to put in our module.

**Review by eeeady — CHANGES_REQUESTED** [review]
We talked... I agree this is relatively harmless since our default will still be to have logs. This will give folks that consume this more options.

File: main.tf [hunk 1]
```diff
@@ -62,10 +62,14 @@ resource "aws_lb" "main" {
   security_groups = [aws_security_group.alb_sg.id]
   idle_timeout    = var.alb_idle_timeout
 
-  access_logs {
-    enabled = true
-    bucket  = var.logs_s3_bucket
-    prefix  = "alb/${var.name}-${var.environment}"
+  dynamic "access_logs" {
+    # Skips creating the block if logs_s3_bucket is empty string
+    for_each = var.logs_s3_bucket == "" ? [] : ["create block"]
```
Comment (LinuxBozo) [inline main.tf:67]: You may be running into https://github.com/hashicorp/terraform/issues/22281

So, I would move the ternary logic to locals under a new var. 
```hcl
locals {
  access_logs_buckets = var.logs_s3_bucket == "" ? [] : [var.logs_s3_bucket]
}
```
Then here, you can use that in your block converted to a map and use the key (which we just also set as the value)
```hcl
dynamic "access_logs" {
  for_each = { for v in local.access_logs_buckets : v => v }
  content {
    enabled = true
    bucket  = access_logs.key
    prefix  = "alb/${var.name}-${var.environment}"
  }
}

Comment (brainsik) [reply main.tf]: Mentioned on Slack, but putting here for history: `terraform apply` works dandy. It's unclear why Terratest is losing it's marbles. For science, I've tried your suggestion. Test is running.

**Comment by brainsik** [comment]
@LinuxBozo @dynamike @eeeady I believe this is ready for a real review now.

File: test/terraform_aws_alb_web_containers_test.go [hunk 2]
```diff
@@ -42,6 +42,69 @@ func TestTerraformAwsAlbWebContainersSimpleHttp(t *testing.T) {
...
+		Error: Provider produced inconsistent final plan
+
+		When expanding the plan for module.alb.aws_lb.main to include new values
+		learned so far during apply, provider "registry.terraform.io/hashicorp/aws"
+		produced an invalid new value for .access_logs[0].bucket: was
+		cty.StringVal(""), but now cty.StringVal("terratest-rbbbb7-logs").
+
+		This is a bug in the provider, which should be reported in the provider's own
+		issue tracker.
+		https://github.com/terraform-providers/terraform-provider-aws/issues/10297
```
Comment (dynamike) [inline test/terraform_aws_alb_web_containers_test.go:61]: This is another bug report specific to AWS access logs and ALBs
https://github.com/hashicorp/terraform-provider-aws/issues/16674

**Review by dynamike — COMMENTED** [review]
Playing around with this locally as @LinuxBozo has so amazingly commented on in the test code, I'm able to reproduce this same bug locally. It only happens when you spin up the logs module and this module from scratch. A second apply passes. Running this problem without logging, like @brainsik, doesn't trigger the bug. 

The bit that I don't understand is it doesn't appear to be a problem with the https://github.com/trussworks/terraform-aws-s3-private-bucket module, which has he same dynamic logic being used here. 

My only request is can we add a Note to the README highlighting this bug, it's a pretty painful one for those using the module with logging for the first time.

**Comment by brainsik** [comment]
Woot! The dynamic block planning problem got fixed in Terraform v0.15.2:
https://github.com/hashicorp/terraform/pull/28424

I'm going to open a new PR with my changes. Apparently the change from `master` → `main` means I can't reopen this.

**Pull request closed without merging**